### ISSUE TITLE
pip cannot install Black with Python 3.9.0 (64 bits)

### ISSUE THREAD

**Issue by parvizfarnia** [psf/black#1832]
I use Black with Elpy in Emacs as my development environment for Python

```
OS: Windows 10 (64 bits)
Emacs version: 26.3 (64 bits)
Python version: 3.9.0 (64 bits)

```
As indicated in the documentation of Black, I just run the following to install Black

```
python -m pip install --upgrade pip
pip install black

```
So far this has always been working with any new version of Python and I've never had any problem. This morning, for the first time, I installed Python 3.9.0. When I tried to install Black, here is the error message that I got:

```
pip install black
Collecting black
  Using cached black-20.8b1.tar.gz (1.1 MB)
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: 'python.exe' 'pip' install --ignore-installed --no-user --prefix 'C:\Users\myaccount\AppData\Local\Temp\pip-build-env-13ky2s3e\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=41.0' setuptools-scm wheel
       cwd: None
  Complete output (7 lines):
  WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A92B50>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/
  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A92460>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/
  WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A927F0>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/
  WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A92670>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/
  WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A926D0>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/
  ERROR: Could not find a version that satisfies the requirement setuptools>=41.0 (from versions: none)
  ERROR: No matching distribution found for setuptools>=41.0
  ----------------------------------------
ERROR: Command errored out with exit status 1: 'python.exe' 'pip' install --ignore-installed --no-user --prefix 'C:\Users\myaccount\AppData\Local\Temp\pip-build-env-13ky2s3e\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools>=41.0' setuptools-scm wheel Check the logs for full command output.

```

Now based on the main page of Black on Github, the required version for Python is 3.6.0+ so I was wondering whether I missed something in the installation procedure or it is the fact that Black doesn't support Python 3.9.0 yet.

Thanks in advance

**Comment by JelleZijlstra** [comment]
`  WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x00000204E7A92460>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/setuptools/`
This sounds like an issue with your network connectivity.

**Comment by parvizfarnia** [comment]
Yes, I can confirm. Reading my previous comment, I noticed that I forgot to mention that I install requried packages via a network protected by a proxy and the pip command is a bit different from what I wrote. I apologize for this error, I think when I was writing the comment I didn't pay attention that I was on my personal computer. 

In order to pass through the proxy, obviously I provide it on the command line, so for example let's say that **MyPackage** is the name of a package that I have to install with pip then here is the command that I run

`pip --proxy http://username:password@proxy_domain_with_port install MyPackage`

This is how I managed to install all required packages for elpy except one: Black

However, after reading your comment, I checked again the proxy part, because I thought if there is a network problem (access problem), in my view, there would be the place to search the problem. And indeed it was. It seems that with python 3.9.0, pip doesn't detect the proxy properly on the command line while trying to install Black. So instead of providing it with pip on the command line I defined two environment variables using Windows command line:

```
set http_proxy=http://username:password@proxy_domain_with_port
set https_proxy=http://username:password@proxy_domain_with_port
```

And then I simply run

`pip install black`

This time it worked like a charm! I really don't have any explanation for this, especially after intalling all other packages required by elpy by providing directly the proxy value as a command line parameter.

Anyway, the problem is solved, I wrote these lines for those whom may have encountered the same problem.

Thanks for the help.
